### PR TITLE
IGAPP-173: Update react and subdependencies

### PR DESCRIPTION
**Pull request by Znippet**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Review by maxammann — APPROVED** [review]
Testing now on iOS. Sombody else should test this on android :)

**Comment by maxammann** [comment]
I pushed some fixes for iOS, more to come.

As Sentry was upgraded we also need to test sentry in this PR.

**Comment by maxammann** [comment]
Also check whether Push Notifications work on iOS (and Android)

**Comment by SteffiStuffel** [comment]
before yarn install worked, i had to delete .cache/yarn in my home-folder.
before app start worked, i had to delete /android/app/build and android/.gradle

App workes, couldn'f find a PDF, so did not test this. Where are pdfs?
Tried to test Push Notifications, got a sound and a little yellow dot on the home screen above the integreat logo. Not sure if this is the desired behaviour. 

Tried to test sentry, but yarn android:integreat:production got stuck

<===========--> 87% EXECUTING [6m 0s]
> :sentry_react-native:generateReleaseRFile
> :app:bundleReleaseJsAndAssets
> IDLE
> :rn-fetch-blob:generateReleaseRFile

File: native/package.json [hunk 1]
```diff
@@ -63,28 +63,28 @@
...
+    "react-native-safe-area-context": "^3.2.0",
     "react-native-safe-area-view": "2.0.0",
-    "react-native-screens": "^2.16.1",
+    "react-native-screens": "^2.18.1",
     "react-native-static-server": "^0.5.0",
     "react-native-svg": "^12.1.0",
-    "react-native-vector-icons": "^7.1.0",
-    "react-native-webview": "10.8.2",
+    "react-native-vector-icons": "^8.1.0",
+    "react-native-webview": "11.2.3",
```
Comment (klinzo) [inline native/package.json:85]: Must not be upgraded as long as this issue is present: https://github.com/react-native-webview/react-native-webview/issues/1741

**Comment by Znippet** [comment]
Sorry for the late response, didn't find any time until now 

Thanks to you all for reviewing and testin!

> "react-native-webview": "11.2.3",
Must not be upgraded as long as this issue is present: react-native-webview/react-native-webview#1741

ok

> before yarn install worked, i had to delete .cache/yarn in my home-folder.
> before app start worked, i had to delete /android/app/build and android/.gradle
> 
> Tried to test sentry, but yarn android:integreat:production got stuck
> oke after 8m it continued, but failed with:
> 
> * What went wrong:
>   Execution failed for task ':app:packageRelease'.
> 
> > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
> > Java heap space

Thanks, I will test it

Also will thave a look at the build failing ("check_native")

File: native/src/routes/offers/containers/__tests__/OffersContainer.spec.js [hunk 2]
```diff
@@ -48,8 +48,8 @@ describe('OffersContainer', () => {
       <OffersContainer navigation={navigation} route={route} />
     )
     expect(getByText('Offers')).toBeTruthy()
-    expect(() => getByText('loading')).toThrow('No instances found with text "loading"')
-    expect(() => getByText(errorText)).toThrow(`No instances found with text "${errorText}"`)
+    expect(() => getByText('loading')).toThrow('Unable to find an element with text: loading')
```
Comment (klinzo) [inline native/src/routes/offers/containers/__tests__/OffersContainer.spec.js:51]: I guess you could just remove the specific error message as well:
```suggestion
    expect(() => getByText('loading')).toThrow()
```

Comment (maxammann) [reply native/src/routes/offers/containers/__tests__/OffersContainer.spec.js:51]: or even better use: expect(() => queryByText('loading')).toBeFalsy().

**Review by maxammann — COMMENTED** [review]
Almost for got this: Always check whether there are changes which need to be applied to the project:

https://react-native-community.github.io/upgrade-helper/?from=0.63.3&to=0.63.4

For this upgrade there aren't any changes as far as i can see.
Here is the view on the project structure how it should be:

**Review by maxammann — APPROVED** [review]

File: native/src/routes/offers/containers/__tests__/OffersContainer.spec.js (hunk 2 above)
Comment (Taggotty) [reply native/src/routes/offers/containers/__tests__/OffersContainer.spec.js:51]: Or even better 🙈 `expect(queryByText('loading')).toBeFalsy()`

File: native/package.json (hunk 1 above)
Comment (Znippet) [reply native/package.json]: done, reverted

File: native/src/routes/offers/containers/__tests__/OffersContainer.spec.js (hunk 2 above)
Comment (Znippet) [reply native/src/routes/offers/containers/__tests__/OffersContainer.spec.js:74]: done

**Comment by Znippet** [comment]
@Taggotty :
> Tried to test sentry, but yarn android:integreat:production got stuck

For me it works, but takes a lot of time:
BUILD SUCCESSFUL in 13m 14s

**Comment by maxammann** [comment]
For some reason this build is failing because of a rnpermissions upgrade: https://github.com/zoontek/react-native-permissions/blob/4b4ec964e7d26278dc00c217ed01ad2830bec3c3/MIGRATION.md#whats-new

Some ios 14 features were added. Somehow this does not compile right now with our project. This means someone has to debug this on a mac.

We sould merge this asap

**Comment by maxammann** [comment]
I bumped xcode to 12 to try whether that fixes the problem

File: native/flow-typed/npm/get-dev-paths_vx.x.x.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// flow-typed signature: 687774148ae4711cc225de5e836d5456
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/flow-typed/npm/get-dev-paths_vx.x.x.js:1]: Actually I think there should be now new type files

File: native/flow-typed/npm/api-client_vx.x.x.js [hunk 4]
```diff
@@ -0,0 +1,692 @@
+// flow-typed signature: f1fa1e545c05e81858d0ff60492b30b8
```
[the remark this answers is not in the document]
Comment (Znippet) [reply native/flow-typed/npm/api-client_vx.x.x.js]: removed

File: native/flow-typed/npm/get-dev-paths_vx.x.x.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Znippet) [reply native/flow-typed/npm/get-dev-paths_vx.x.x.js]: All flow types with "any" are removed 
I also reverted the old ones wich where automatically deleted by flow-typed as I saw that the versions mostly seem to be compatible

**Review by klinzo — APPROVED** [review]
Tested on Android :+1: Good to merge from my side :)

**Pull request merged**